build-logic/build.gradle.kts

```diff
@@ -1,15 +1,18 @@
+buildscript {
+    dependencies {
+        with(libs.kotlin.gradle.get()) {
+            classpath("$group:$name:$embeddedKotlinVersion")
+        }
+    }
+}
+
 plugins {
     `kotlin-dsl`
 }
 
 dependencies {
-    implementation(libs.kotiln.gradle)
+    implementation(libs.kotlin.gradle)
     implementation(libs.detekt.gradle)
-    // hack to access version catelouge https://github.com/gradle/gradle/issues/15383
+    // hack to access version catalogue https://github.com/gradle/gradle/issues/15383
     implementation(files(libs.javaClass.superclass.protectionDomain.codeSource.location))
 }
-
-java {
```

Comment:
note: to run with java 20 I had to add
```
java {
    targetCompatibility = JavaVersion.VERSION_11
}

tasks.withType<KotlinCompile>().configureEach {
    compilerOptions.jvmTarget = JvmTarget.JVM_11
}
```
(17 probably would work as well)
but either way I think it's good to have for consistency.

Reply:
Using a toolchain here would probably be a valid use case (and I tested that it works), but the alternative is to not use java 20 until Gradle supports it.

Reply:
> but the alternative is to not use java 20 until Gradle supports it

yeah true, I might end up adding a target here because it just means you'd be generating older class files than needed so it's not a huge issue, but it's not a priority or anything.